lib/utils.js

```diff
@@ -689,22 +689,27 @@ function* makeCounter(seed) {
 /**
  * Helper function for either accepting a callback, or returning a promise
  *
- * @param {Function} [callback] an optional callback.
- * @param {Function} fn A function that takes a callback
+ * @param {Object} parent an instance of parent with promiseLibrary.
+ * @param {object} parent.s an object containing promiseLibrary.
+ * @param {function} parent.s.promiseLibrary an object containing promiseLibrary.
+ * @param {function | undefined} callback The last function argument in exposed method, controls if a Promise is returned
+ * @param {function} operation A function that wraps the callback
```

Comment:
I renamed `fn` to `operation` because every time I look at this I get confused as to why `callback` isn't the last argument, and it not being clear which argument function is coming from where.

Reply:
I'd prefer we not use the term "operation" here because we already have such a concept. If `fn` is undesirable, maybe you can call it `body` or `lambda` or something else

Reply:
How about `wrapper` since the jsdoc says it is a function that "wraps" the callback?